src/SMatrix.jl

```diff
@@ -53,9 +53,12 @@ end
     end
 end
 
-@inline convert{S1,S2,T}(::Type{SMatrix{S1,S2}}, a::StaticArray{T}) = SMatrix{S1,S2,T}(Tuple(a))
+@inline convert{S1,S2,T}(::Type{SMatrix{S1,S2}}, a::StaticArray{<:Any, T}) = SMatrix{S1,S2,T}(Tuple(a))
```

Comment:
It does allow reshaping... but primarily it provides the functionality to change `T`.

Should we tighten this up at some point?